master/internal/trial.go

```diff
@@ -452,6 +452,21 @@ func (t *trial) maybeAllocateTask() error {
 		Debugf("starting new trial allocation")
 
 	prom.AssociateJobExperiment(t.jobID, strconv.Itoa(t.experimentID), t.config.Labels())
+
+	// TODO: The first return value is []command.LaunchWarning{command.CurrentSlotsExceeded}.
+	// How do we want to handle this? In newExperiment() the warning is part of the response back
+	// to the client.
+	_, err = t.rm.ValidateResourcePoolAvailability(
```

Comment:
I think we don't want to run this in the restore case

Reply:
For understanding, what's the issue if we run this in the restore case?

Reply:
We want skip the restore case because I think there is a possibility not all agents will be reconnected when the job is restoring. 

In addition, really the only time we want this to fail is when a trial decides to reschedule and finds it can't.